dynamic_rest/serializers.py

```diff
@@ -301,6 +317,29 @@ def get_name(cls):
         return cls.Meta.name
 
     @classmethod
+    def get_url(self, pk=None):
+        # if associated with a registered viewset, use its URL
+        url = getattr(self, '_url', None)
```

Comment:
Right; I think there's still value in the `register_resource` API in that it helps integration code outside of the API link arbitrary instances or models back to the API. 

However, I do think its worth examining the corner cases surfaced here and attempting to simplify. So far, here's what I'm hearing:

1) A single model can be represented by multiple serializers and endpoints, e.g. `/v1/users/` and `/v2/users/` 
2) A single serializer can be re-used across endpoints, e.g. `/v1/users/` and `/v1/people/` (different views of the same data)
3) A view can use different serializers depending on the request context, e.g. an admin version and a basic version (this is a DRF example)

This makes things very complicated in the worst case: models/serializers/views are all many-to-many. I think we should consider officially dropping support for 2) and 3) and arrive at e.g:

- A viewset has exactly one serializer and at most one model (ephemeral serializers have no model)
- A serializer has at most one viewset and at most one model (ephemeral serializers have no model, some serializers may not be exposed directly)
- A model can be represented by many serializers/viewsets